cpp/src/arrow/compute/kernels/aggregate_var_std.cc

```diff
@@ -60,16 +60,19 @@ struct VarStdState {
     if (count == 0 || (!this->all_valid && !options.skip_nulls)) {
       return;
     }
-
+    Status status;
     using SumType = typename internal::GetSumType<T>::SumType;
-    SumType sum = internal::SumArray<CType, SumType, SimdLevel::NONE>(array);
+    SumType sum =
+        internal::SumArray<CType, SumType, SimdLevel::NONE, false>(array, &status);
 
     const double mean = ToDouble(sum) / count;
-    const double m2 = internal::SumArray<CType, double, SimdLevel::NONE>(
-        array, [this, mean](CType value) {
+    const double m2 = internal::SumArray<CType, double, SimdLevel::NONE, false>(
+        array,
+        [this, mean](CType value) {
           const double v = ToDouble(value);
           return (v - mean) * (v - mean);
-        });
+        },
+        &status);
```

Comment:
Let's make sure we don't let an error slip here? (it shouldn't happen now but perhaps it will at some point?)
```suggestion
        &status);
    ARROW_CHECK_OK(status);
```

Reply:
Done